modules/nextrollBidAdapter.js

```diff
@@ -0,0 +1,234 @@
+   * @return {UserSync[]} The user syncs which should be dropped.
+   */
+  getUserSyncs: function (syncOptions, serverResponses, gdprConsent) {
+    return [];
+  }
+}
+
+function _getUser(requests) {
+  let id = utils.deepAccess(requests, '0.userId.nextroll');
+  if (id === undefined) {
+    return
+  }
+
+  return {
+    ext: {
+      eid: [{
+        'source': 'nextroll',
+        id
+      }]
+    }
+  }
+}
+
+function _buildResponse(bidResponse, bid) {
+  const adm = utils.replaceAuctionPrice(bid.adm, bid.price);
+  return {
+    requestId: bidResponse.id,
+    cpm: bid.price,
+    width: bid.w,
+    height: bid.h,
+    creativeId: bid.crid,
+    dealId: bidResponse.dealId,
+    currency: 'USD',
+    netRevenue: true,
+    ttl: 300,
+    ad: adm
+  }
+}
+
+function _getSite(bidRequest, topLocation) {
+  return {
+    page: topLocation.href,
+    domain: topLocation.hostname,
+    publisher: {
+      id: utils.getBidIdParameter('publisherId', bidRequest.params)
+    }
+  }
+}
+
+function _getSeller(bidRequest) {
+  return {
+    id: utils.getBidIdParameter('sellerId', bidRequest.params)
+  }
+}
+
+function _getSizes(bidRequest) {
+  return bidRequest.sizes.filter(_isValidSize).map(size => {
+    return {
+      w: size[0],
+      h: size[1]
+    }
+  })
+}
+
+function _isValidSize(size) {
+  const isNumber = x => typeof x === 'number';
+  return (size.length === 2 && isNumber(size[0]) && isNumber(size[1]));
+}
+
+function _getDevice(_bidRequest) {
+  return {
+    ua: navigator.userAgent,
+    language: navigator['language'],
+    os: _getOs(navigator.userAgent.toLowerCase()),
+    osv: _getOsVersion(navigator.userAgent)
+  }
+}
+
+function _getOs(userAgent) {
+  const osTable = {
+    'android': /android/i,
+    'ios': /iphone|ipad/i,
+    'mac': /mac/i,
+    'linux': /linux/i,
+    'windows': /windows/i
+  };
+
+  return find(Object.keys(osTable), os => {
+    if (userAgent.match(osTable[os])) {
+      return os;
+    }
+  }) || 'etc';
+}
+
+function _getOsVersion(userAgent) {
+  let clientStrings = [
+    { s: 'Android', r: /Android/ },
+    { s: 'iOS', r: /(iPhone|iPad|iPod)/ },
+    { s: 'Mac OS X', r: /Mac OS X/ },
+    { s: 'Mac OS', r: /(MacPPC|MacIntel|Mac_PowerPC|Macintosh)/ },
+    { s: 'Linux', r: /(Linux|X11)/ },
+    { s: 'Windows 10', r: /(Windows 10.0|Windows NT 10.0)/ },
+    { s: 'Windows 8.1', r: /(Windows 8.1|Windows NT 6.3)/ },
+    { s: 'Windows 8', r: /(Windows 8|Windows NT 6.2)/ },
+    { s: 'Windows 7', r: /(Windows 7|Windows NT 6.1)/ },
+    { s: 'Windows Vista', r: /Windows NT 6.0/ },
+    { s: 'Windows Server 2003', r: /Windows NT 5.2/ },
+    { s: 'Windows XP', r: /(Windows NT 5.1|Windows XP)/ },
+    { s: 'UNIX', r: /UNIX/ },
+    { s: 'Search Bot', r: /(nuhk|Googlebot|Yammybot|Openbot|Slurp|MSNBot|Ask Jeeves\/Teoma|ia_archiver)/ }
+  ];
+  let cs = find(clientStrings, cs => cs.r.test(userAgent));
+  return cs ? cs.s : 'unknown';
+}
+
+function _parseUrl(url) {
```

Comment:
Can you use functions from https://github.com/prebid/Prebid.js/blob/master/src/url.js